README.md
### aimsir17
Irish Weather Data from Ireland's [Weather Observing Stations](https://www.met.ie/climate/weather-observing-stations) in 2017. It now includes energy generation data for 2017 in the tibble *eirgrid17*.

Hourly records from January-December based on the following weather observation stations:

- Athenry, Co. Galway
- Ballyhaise, Co. Cavan
- Belmullet, Co. Mayo
- Casement Aerodrome, Co. Dublin
- Claremorris, Co. Mayo
- Cork Airport, Co. Cork
- Dublin Airport, Co. Dublin
- Dunsany (Grange), Co. Meath
- Finner Camp, Co. Donegal
- Gurteen, Co. Tipperary
- Johnstown Castle, Co. Wexford
- Knock Airport, Co. Mayo
- Mace Head, Co. Galway
- Malin Head, Co. Donegal
- Markree Castle, Co. Sligo
- Moore Park, Co. Cork
- Mount Dillon, Co. Roscommon
- Mullingar, Co. Westmeath
- Newport Furnace, Co. Mayo
- Oak Park, Co. Carlow
- Phoenix Park, Co. Dublin
- Roches point SWS, Co. Cork
- Shannon Airport, Co. Clare
- Sherkin Island, Co. Cork
- Valentia Observatory, Co. Kerry

See also, [major weather events](https://www.met.ie/climate/major-weather-events), which in 2017 including [Storm Ophelia](https://en.wikipedia.org/wiki/Hurricane_Ophelia_(2017)) (October 9, 2017 – October 18, 2017)

### Examples
In order to maximise the usefulness of the observations, R's [tidyverse](https://www.tidyverse.org) tools such as **dplyr** and **ggplot2** can be used. Here are three examples showing their use.

- [Loading the weather data](
https://github.com/JimDuggan/aimsir17/tree/master/data-raw/Examples/01%20Loading%20Dataset)
- [Exploring a weather event: Storm Ophelia](
https://github.com/JimDuggan/aimsir17/tree/master/data-raw/Examples/02%20Storm%20Ophelia)
- [Summarising hourly rainfall into monthly values](
https://github.com/JimDuggan/aimsir17/tree/master/data-raw/Examples/03%20Rainfall%20Summary)
- [Exploring the eirgrid energy data](
https://github.com/JimDuggan/aimsir17/tree/master/data-raw/Examples/04%20Energy%20Data)
- [Joining weather and energy data](
https://github.com/JimDuggan/aimsir17/tree/master/data-raw/Examples/05%20Joining%20Data)

### Author Information

- Author: Jim Duggan
- Affiliations
  + [School of Computer Science, NUI Galway.](https://www.nuigalway.ie/engineering-informatics/information-technology/)
  + [Ryan Institute, NUI Galway.](http://www.nuigalway.ie/ryaninstitute/)
  + [Insight Centre for Data Analytics, NUI Galway.](https://www.insight-centre.org)

### Acknowledgement

- Copyright Met Éireann and EirGrid
- Source [www.met.ie](http://www.met.ie), http://www.eirgridgroup.com/how-the-grid-works/renewables/
- This data is published under a [Creative Commons Attribution 4.0 International (CC BY 4.0)]( https://creativecommons.org/licenses/by/4.0/)
- Met Éireann does not accept any liability whatsoever for any error or omission in the data, their availability, or for any loss or damage arising from their use.
- This material has been modified from the original, in terms of the following:
  + New columns added to the data for Year, Month, Day and Hour. 
  + Columns representing indicators removed.
  + The column date transformed to a `POSIXct` type
- For the energy data from EirGrid, please note that the data contained herein is indicative as it is based on raw 15-minute SCADA readings which have yet to be quality controlled. EirGrid as the Transmission System Operator (TSO) for Ireland and SONI as the TSO for Northern Ireland make no warranties or representations of any kind with respect of this document, including, without limitation, its quality, accuracy and completeness.
While every effort has been made in the compilation of this report to ensure that the information
herein is correct, the TSOs do not accept liability for any loss or damage arising from the use of this
document or any reliance on the information it contains. Use of this document and the information it
contains is at the user’s sole risk. In addition, the TSOs strongly recommend that any party wishing
to make a decision based on the content of this document should consult the TSOs in advance.


NEWS.md
### aimsir17 Version 0.0.1
This is the first release of this data package containing Irish Weather Observing Stations Hourly Records for 2017.

- Added copyright information on 28/11/2019, referencing data sources as Met Eireann.


### aimsir17 Version 0.0.2
This is the second release of this data package containing Irish Weather Observing Stations Hourly Records for 2017. New data has been added relating to energy demand and generation from the Irish Grid for 2017. This data can be joined with the existiong weather data.



R/stations.R
#' Summary of weather observing stations with observations
#'
#' @format
#' \describe{
#' \item{station}{Name of the Weather Observing Station}
#' \item{height}{Height above sea level (M)}
#' \item{latitude}{Latitude of weather station}
#' \item{longitude}{Longitude of weather station}
#' \item{Copyright}{Met Eireann}
#' \item{Licence Statement}{This data is published under a Creative Commons Attribution 4.0 International (CC BY 4.0).
#' https://creativecommons.org/licenses/by/4.0/}
#' \item{Disclaimer}{Met Eireann does not accept any liability whatsoever for any error or omission in the data, their availability,
#' or for any loss or damage arising from their use}
#' #' \item{Modifications}{New columns added to the data for Year, Month, Day and Hour. All ind columns removed.
#' Date formatted to R data object format using package lubridate}
#' }
#' @source \url{ https://www.met.ie}
"stations"





R/eirgrid17.R
#' EirGrid System Data Quarterly Hourly for 2017
#'
#'
#' @format
#' \describe{
#' \item{year}{Year of observation.}
#' \item{month}{Month of observation.}
#' \item{day}{Day of observation.}
#' \item{hour}{Hour of observation.}
#' \item{minute}{Minute of observation.}
#' \item{date}{Date and Time (utc).}
#' \item{NIGeneration}{Northern Ireland Generation.}
#' \item{NIDemand}{Northern Ireland Demand.}
#' \item{NIWindAvailability}{Northern Ireland Wind Availability.}
#' \item{NIWindGeneration}{Northern Ireland Wind Generation.}
#' \item{IEGeneration}{Republic of Ireland Generation.}
#' \item{IEDemand}{Republic of Ireland Demand.}
#' \item{IEWindAvailability}{Republic of Ireland Wind Availability.}
#' \item{NIWindGeneration}{Republic of Irekand Wind Generation.}
#' \item{SNSP}{System Non-Synchronous Penetration) is the sum of wind + solar generation + HVDC imports as a percentage of total demand + exports}
#' \item{Copyright}{EirGrid}
#' \item{Licence Statement}{This data is published under a Creative Commons Attribution 4.0
#' International (CC BY 4.0). https://creativecommons.org/licenses/by/4.0/}
#' \item{Disclaimer}{Disclaimer: Please note that the data contained herein is indicative as it is based on raw 15-minute SCADA readings
#'                               which have yet to be quality controlled. EirGrid as the Transmission System Operator (TSO) for Ireland
#'                               and SONI as the TSO for Northern Ireland make no warranties or representations of any kind with respect
#'                               of this document, including, without limitation, its quality, accuracy and completeness.
#'                               While every effort has been made in the compilation of this report to ensure that the information
#'                               herein is correct, the TSOs do not accept liability for any loss or damage arising from the use of this
#'                               document or any reliance on the information it contains. Use of this document and the information it
#'                               contains is at the user’s sole risk. In addition, the TSOs strongly recommend that any party wishing
#'                               to make a decision based on the content of this document should consult the TSOs in advance. }
#' \item{Modifications}{New columns added to the data for Year, Month, Day, Hour and Minute. Columns representing GMT offset removed.
#' The column date transformed to a `POSIXct` type. Furthermore, note that:
#' The system data in this report is is based on average 15-minute SCADA readings (MW).
#' All figures represent net exported energy.
#' Maximum/Minimum figures in the different jurisdictions do not necessarily occur at the same time.
#' The maximum SNSP limit allowed on the system is 65%.
#' SNSP may exceed this maximum limit due to wind variations.}
#' }
#' @source \url{http://www.eirgridgroup.com/how-the-grid-works/renewables/}
"eirgrid17"

#' @importFrom tibble tibble
NULL













R/observations.R
#' Met Eireann - Weather Observing Stations Records 01-Jan-2017 to 31-Dec-2017
#'
#'
#' @format
#' \describe{
#' \item{date}{Date and Time (utc).}
#' \item{rain}{Precipitation Amount (mm).}
#' \item{temp}{Air Temperature (C)}
#' \item{wetb}{Wet Bulb Temperature (C)}
#' \item{dewpt}{Dew Point Temperature (C)}
#' \item{vappr}{Vapour Pressure (hPa)}
#' \item{rhum}{Relative Humidity \% }
#' \item{msl}{Mean Sea Level Pressure (hPa)}
#' \item{wddir}{Predominant Wind Direction (deg)}
#' \item{Copyright}{Met Eireann}
#' \item{Licence Statement}{This data is published under a Creative Commons Attribution 4.0
#' International (CC BY 4.0). https://creativecommons.org/licenses/by/4.0/}
#' \item{Disclaimer}{Met Eireann does not accept any liability whatsoever for any error or omission in the data, their availability,
#' or for any loss or damage arising from their use}
#' \item{Modifications}{New columns added to the data for Year, Month, Day and Hour. Columns representing indicators removed.
#' The column date transformed to a `POSIXct` type}
#' }
#' @source \url{https://www.met.ie}
"observations"

#' @importFrom tibble tibble
NULL




data-raw/prepare.R
library(readr)
library(lubridate)
library(dplyr)
library(devtools)

START_YEAR  <- 2017
FINISH_YEAR <- 2017

stations <- tibble(
  station = character(),
  county  = character(),
  height = numeric(),
  latitude = numeric(),
  longitude=numeric()
)

# Removing some vars to save on space...
prepare <- function(file,station,l_skip=17){
  f <- read_csv(file,skip=l_skip)
  # add a column for wind speed if it is missing
  if(!("wdsp" %in% names(f))){
    f <- mutate (f,wdsp=NA)
  }

  if(!("wddir" %in% names(f))){
    f <- mutate (f,wddir=NA)
  }

  f %>% select(date,rain,temp,-wetb,-dewpt,-vappr,rhum,msl,wdsp,wddir,-starts_with("ind")) %>%
        mutate(date=dmy_hm(date),
              year=year(date),
              month=month(date),
              day=day(date),
              hour=hour(date),
              station=station)    %>%
       select(station, year, month, day, hour, everything()) %>%
       filter(year >= START_YEAR, year <= FINISH_YEAR)
}

eirgrid <- function(){
  d <- read_csv("data-raw/EirGrid/System-Data-Qtr-Hourly-2017.csv") %>%
               select(-`GMT Offset`) %>%
               rename(date=DateTime,
                      NIGeneration=`NI Generation`,
                      NIDemand=`NI Demand`,
                      NIWindAvailability=`NI Wind Availability`,
                      NIWindGeneration=`NI Wind Generation`,
                      IEGeneration=`IE Generation`,
                      IEDemand=`IE Demand`,
                      IEWindAvailability=`IE Wind Availability`,
                      IEWindGeneration=`IE Wind Generation`
                      ) %>%
               mutate(date=dmy_hm(date),
                      year=year(date),
                      month=month(date),
                      day=day(date),
                      hour=hour(date),
                      minute=minute(date)) %>%
                      select(year, month, day, hour, minute, everything())

  d

}

# (1) Athenry weather observing station
athenry   <- prepare("data-raw/original data/athenry/hly1875/hly1875.csv","ATHENRY")
stations  <- add_row(stations,station="ATHENRY",county="Galway",height=40,latitude=53.289,longitude=-8.786)

# (2) Ballyhaise weather observing station
ballyhaise <- prepare("data-raw/original data/ballyhaise/hly675/hly675.csv","BALLYHAISE")
stations   <- add_row(stations,station="BALLYHAISE",county="Cavan",height=78,latitude=54.051,longitude=-7.310)

# (3) Belmullet weather observing station
belmullet <- prepare("data-raw/original data/belmullet/hly2375/hly2375.csv","BELMULLET",l_skip=22)
stations   <- add_row(stations,station="BELMULLET",county="Mayo",height=9,latitude=54.228,longitude=-10.007)

# (4) Casement
casement <- prepare("data-raw/original data/casement aerodrome/hly3723/hly3723.csv","CASEMENT",l_skip = 23)
stations <- add_row(stations,station="CASEMENT",county="Dublin",height=91,latitude=53.306,longitude=-6.439)


# (5) Claremorris
claremorris <- prepare("data-raw/original data/claremorris/hly2175/hly2175.csv","CLAREMORRIS")
stations <- add_row(stations,station="CLAREMORRIS",county="Mayo",height=68,latitude=53.711,longitude=-8.993)

# (6) Cork Airport
cork <- prepare("data-raw/original data/cork airport/hly3904/hly3904.csv","CORK AIRPORT",l_skip = 23)
stations <- add_row(stations,station="CORK AIRPORT",county="Cork",height=155,latitude=51.847 ,longitude=-8.486)

# (7) Dublin Airport
dublin <- prepare("data-raw/original data/dublin airport/hly532/hly532.csv","DUBLIN AIRPORT",l_skip = 23)
stations <- add_row(stations,station="DUBLIN AIRPORT",county="Dublin",height=71,latitude=53.428 ,longitude=-6.241)

# (8) Dunsany
dunsany <- prepare("data-raw/original data/dunsany/hly1375/hly1375.csv","DUNSANY")
stations <- add_row(stations,station="DUNSANY",county="Meath",height=83,latitude=53.516,longitude=-6.660)

# (9) Finner
finner <- prepare("data-raw/original data/finner/hly2075/hly2075 fix2.csv","FINNER",l_skip=0)
stations <- add_row(stations,station="FINNER",county="Donegal",height=33,latitude=54.494,longitude=-8.243)

# (10) Gurteen
gurteen <- prepare("data-raw/original data/gurteen/hly1475/hly1475.csv","GURTEEN")
stations <- add_row(stations,station="GURTEEN",county="Tipperary",height=75,latitude=53.053,longitude=-8.009)

# (11) Johnstown Castle
j_castle <- prepare("data-raw/original data/johnstown castle/hly1775/hly1775.csv","JOHNSTOWNII")
stations <- add_row(stations,station="JOHNSTOWNII",county="Wexford",height=62,latitude=52.298,longitude=-6.497)

# (12) Knock
knock <- prepare("data-raw/original data/knock airport/hly4935/hly4935.csv","KNOCK AIRPORT",l_skip = 23)
stations <- add_row(stations,station="KNOCK AIRPORT",county="Mayo",height=201,latitude=53.906,longitude=-8.817)

# (13) Mace Head weather observing station
mace_head <- prepare("data-raw/original data/mace head/hly275.csv","MACE HEAD")
stations  <- add_row(stations,station="MACE HEAD",county="Galway",height=21,latitude=53.326,longitude=-9.901)


# (14) Malin Head weather observing station
malin_head <- prepare("data-raw/original data/malin head/hly1575/hly1575.csv","MALIN HEAD",l_skip=23)
stations  <- add_row(stations,station="MALIN HEAD",county="Donegal",height=20,latitude=55.372,longitude=-7.339)

# (15) Markree weather observing station
markree <- prepare("data-raw/original data/markree/hly1275/hly1275.csv","MARKREE",l_skip = 15)
stations  <- add_row(stations,station="MARKREE",county="Sligo",height=34,latitude=54.175,longitude=-8.456)

# (16) Moore Park (Fermoy)
moore_park <- prepare("data-raw/original data/fermoy/hly575/hly575.csv","MOORE PARK")
stations <- add_row(stations,station="MOORE PARK",county="Cork",height=46,latitude=52.164,longitude=-8.264)

# (17) Mount Dillon
mt_dillon <- prepare("data-raw/original data/mount dillon/hly1975/hly1975.csv","MT DILLON")
stations  <- add_row(stations,station="MT DILLON",county="Roscommon",height=39,latitude=53.727 ,longitude=-7.981)

# (18)  Mullingar
mullingar <- prepare("data-raw/original data/mullingar/hly875/hly875.csv","MULLINGAR")
stations  <- add_row(stations,station="MULLINGAR",county="Westmeath",height=101,latitude=53.537 ,longitude=-7.362)

# (19)  Newport
newport <- prepare("data-raw/original data/newport/hly1175/hly1175.csv","NEWPORT")
stations  <- add_row(stations,station="NEWPORT",county="Mayo",height=22,latitude=53.922 ,longitude=-9.572)

# (20) Oak Park
oak_park <- prepare("data-raw/original data/carlow/hly375/hly375.csv","OAK PARK")
stations <- add_row(stations,station="OAK PARK",county="Carlow",height=62,latitude=52.861,longitude=-6.915)

# (21) Pheonix Park
pheonix_park <- prepare("data-raw/original data/pheonix park/hly175/hly175.csv","PHOENIX PARK",l_skip = 15)
stations <- add_row(stations,station="PHOENIX PARK",county="Dublin",height=48,latitude=53.364,longitude=-6.350)

# (22) Roches Point
roches_point <- prepare("data-raw/original data/roches point/hly1075/hly1075.csv","ROCHES POINT")
stations <- add_row(stations,station="ROCHES POINT",county="Cork",height=40,latitude=51.793,longitude=-8.244)

# (23) Shannon Airport
shannon <- prepare("data-raw/original data/shannon airport/hly518/hly518.csv","SHANNON AIRPORT",l_skip = 23)
stations <- add_row(stations,station="SHANNON AIRPORT",county="Clare",height=15,latitude=52.690,longitude=-8.918)

# (24) Sherkin Island
sherkin <- prepare("data-raw/original data/sherkin island/hly775/hly775.csv","SherkinIsland")
stations <- add_row(stations,station="SherkinIsland",county="Cork",height=21,latitude=51.476,longitude=-9.428)

# (25) Valentia
valentia<- prepare("data-raw/original data/valentia/hly2275/hly2275.csv","VALENTIA OBSERVATORY",l_skip = 23)
stations <- add_row(stations,station="VALENTIA OBSERVATORY",county="Kerry",height=24,latitude=51.938,longitude=-10.241)


observations <- athenry                  %>%       # (01)
                bind_rows(ballyhaise)    %>%       # (02)
                bind_rows(belmullet)     %>%       # (03)
                bind_rows(casement)      %>%       # (04)
                bind_rows(claremorris)   %>%       # (05)
                bind_rows(cork)          %>%       # (06)
                bind_rows(dublin)        %>%       # (07)
                bind_rows(dunsany)       %>%       # (08)
                bind_rows(finner)        %>%       # (09)
                bind_rows(gurteen)       %>%       # (10)
                bind_rows(j_castle)      %>%       # (11)
                bind_rows(knock)         %>%       # (12)
                bind_rows(mace_head)     %>%       # (13)
                bind_rows(malin_head)    %>%       # (14)
                bind_rows(markree)       %>%       # (15)
                bind_rows(moore_park)    %>%       # (16)
                bind_rows(mt_dillon)     %>%       # (17)
                bind_rows(mullingar)     %>%       # (18)
                bind_rows(newport)       %>%       # (19)
                bind_rows(oak_park)      %>%       # (20)
                bind_rows(pheonix_park)  %>%       # (21)
                bind_rows(roches_point)  %>%       # (22)
                bind_rows(shannon)       %>%       # (23)
                bind_rows(sherkin)       %>%       # (24)
                bind_rows(valentia)                # (25)

use_data(observations,overwrite = T)

# Save Stations tibble
use_data(stations,overwrite = T)

# Check rainfall records
check_year <- observations %>%
            group_by(station) %>%
            summarise(Records =n(),
                      MissingRecords=sum(is.na(rain)),
                      TotalRain=sum(rain,na.rm=T)) %>%
            arrange(station)


check_month <- observations %>%
  group_by(month,station) %>%
  summarise(Records =n(),
            MissingRecords=sum(is.na(rain)),
            TotalRain=sum(rain,na.rm=T)) %>%
  arrange(station)

check_data <- select(check_month,station,month,TotalRain)

check_wide <- pivot_wider(check_data,names_from = month,values_from = TotalRain)

# To test values, see https://www.met.ie/climate/available-data/monthly-data

# August 2020, Adding Eirgrid data

eirgrid17 <- eirgrid()

use_data(eirgrid17,overwrite = T)



